holoviews/core/accessors.py

```diff
@@ -0,0 +1,463 @@
+"""
+Module for accessor objects for viewable HoloViews objects.
+"""
+from __future__ import absolute_import, unicode_literals
+
+from collections import OrderedDict
+
+import param
+
+from . import util
+from .pprint import PrettyPrinter
+
+
+class Apply(object):
+    """
+    Utility to apply a function or operation to all viewable elements
+    inside the object.
+    """
+
+    def __init__(self, obj, mode=None):
+        self._obj = obj
+
+    def __call__(self, function, streams=[], link_inputs=True, dynamic=None, **kwargs):
+        """Applies a function to all (Nd)Overlay or Element objects.
+
+        Any keyword arguments are passed through to the function. If
+        keyword arguments are instance parameters, or streams are
+        supplied the returned object will dynamically update in
+        response to changes in those objects.
+
+        Args:
+            function: A callable function
+                The function will be passed the return value of the
+                DynamicMap as the first argument and any supplied
+                stream values or keywords as additional keyword
+                arguments.
+            streams (list, optional): A list of Stream objects
+                The Stream objects can dynamically supply values which
+                will be passed to the function as keywords.
+            link_inputs (bool, optional): Whether to link the inputs
+                Determines whether Streams and Links attached to
+                original object will be inherited.
+            dynamic (bool, optional): Whether to make object dynamic
+                By default object is made dynamic if streams are
+                supplied, an instance parameter is supplied as a
+                keyword argument, or the supplied function is a
+                parameterized method.
+            kwargs (dict, optional): Additional keyword arguments
+                Keyword arguments which will be supplied to the
+                function.
+
+        Returns:
+            A new object where the function was applied to all
+            contained (Nd)Overlay or Element objects.
+        """
```

Comment:
Given that the only state used in `__call__` is `self._obj` I would consider pulling this out into a utility with a signature `apply(obj, function, streams=[], link_inputs=True, dynamic=None, **kwargs)`...then this method would just be `self.apply(obj, function, streams=[], link_inputs=True, dynamic=None, **kwargs)`.

This method is also a tad long but I think it is ok as I don't see an obvious way to break it up.

Reply:
Could you explain why you'd prefer that? I don't think users should use it directly and ``redim`` and ``Opts`` accessors don't do this although they also just use ``self._obj`` as state.

Reply:
The utility would also live in this file, so it doesn't help any from an organization standpoint either.

Reply:
> ... although they also just use self._obj as state.

That isn't entirely true, unlike `Apply` they have `self.mode`. The reason I like utility functions is they are more functional - there is nowhere for state to hide outside the call signature. However, I accepts your argument that the other accessors don't do this and now that this code is in a new file anyway, I guess I take back that suggestion.